Code/Mantid/Framework/DataHandling/src/LoadSpiceXML2DDet.cpp

```diff
@@ -0,0 +1,432 @@
+/** Destructor
+ */
+LoadSpiceXML2DDet::~LoadSpiceXML2DDet() {}
+
+//----------------------------------------------------------------------------------------------
+const std::string LoadSpiceXML2DDet::name() const {
+  return "LoadSpiceXML2DDet";
+}
+
+//----------------------------------------------------------------------------------------------
+int LoadSpiceXML2DDet::version() const { return 1; }
+
+//----------------------------------------------------------------------------------------------
+const std::string LoadSpiceXML2DDet::category() const { return "DataHandling"; }
+
+//----------------------------------------------------------------------------------------------
+const std::string LoadSpiceXML2DDet::summary() const {
+  return "Load 2-dimensional detector data file in XML format from SPICE. ";
+}
+
+//----------------------------------------------------------------------------------------------
+/** Declare properties
+ * @brief LoadSpiceXML2DDet::init
+ */
+void LoadSpiceXML2DDet::init() {
+  std::vector<std::string> xmlext;
+  xmlext.push_back(".xml");
+  declareProperty(
+      new FileProperty("Filename", "", FileProperty::FileAction::Load, xmlext),
+      "XML file name for one scan including 2D detectors counts from SPICE");
+
+  declareProperty(new WorkspaceProperty<MatrixWorkspace>("OutputWorkspace", "",
+                                                         Direction::Output),
+                  "Name of output matrix workspace. ");
+
+  declareProperty("DetectorLogName", "Detector",
+                  "Log name for detector counts.");
+
+  declareProperty(
+      new ArrayProperty<size_t>("DetectorGeometry"),
+      "A size-2 unsigned integer array [X, Y] for detector geometry. "
+      "Such that the detector contains X x Y pixels.");
+}
+
+//----------------------------------------------------------------------------------------------
+/** Main execution
+ * @brief LoadSpiceXML2DDet::exec
+ */
+void LoadSpiceXML2DDet::exec() {
+  // Load input
+  const std::string xmlfilename = getProperty("Filename");
+  const std::string detlogname = getProperty("DetectorLogName");
+  std::vector<size_t> vec_pixelgeom = getProperty("DetectorGeometry");
+  if (vec_pixelgeom.size() != 2) {
+    throw std::runtime_error("Input pixels geometry is not correct in format.");
+  }
+  size_t numpixelX = vec_pixelgeom[0];
+  size_t numpixelY = vec_pixelgeom[1];
+
+  // Parse
+  std::vector<SpiceXMLNode> vec_xmlnode;
+  parseSpiceXML(xmlfilename, vec_xmlnode);
+
+  // Create output workspace
+  MatrixWorkspace_sptr outws =
+      createMatrixWorkspace(vec_xmlnode, numpixelX, numpixelY, detlogname);
+
+  setProperty("OutputWorkspace", outws);
+}
+
+//----------------------------------------------------------------------------------------------
+/** Parse SPICE XML file for one Pt./measurement
+ * @brief LoadSpiceXML2DDet::parseSpiceXML
+ * @param xmlfilename
+ * @param logstringmap
```

Comment:
Yes, you are right on this.  I forgot to update the parameter's name after I changed the input from map to vector.  It has been corrected.